searchguard/rolesmapping.py

```diff
@@ -152,3 +165,26 @@ def modify_rolemapping(role, properties, action="replace"):
 
     # No merge or split action, overwrite existing properties:
     _send_api_request(role, properties)
+
+
+def list_rolemappings_for_user(user, roles=None):
+    """Get list of rolemappings that contain the given user. If no roles are provided,
+    all rolemappings are checked.
+
+    :param str user: Name of user
+    :param list roles: List of rolemappings to be checked for the given user
+    :returns list: list of rolemappings with the given user
+    :raises: ViewRoleMappingException
+    """
+    user_rolemappings = []
+
+    if roles:
+        for role in roles:
+            if user in view_rolemapping(role)[role]['users']:
+                user_rolemappings.append(role)
+    else:
+        for role, properties in view_all_rolemappings().items():
```

Comment:
you could also do something like this here: `user_rolemappings = [r for r, p in view_all_rolemappings().items() if user in properties['users']]`